msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -784,7 +784,16 @@ public void acquireTokenSilentAsync(@NonNull final String[] scopes,
         String requestAuthority = authority;
 
         if (StringUtil.isEmpty(requestAuthority)) {
-            requestAuthority = Authority.getAuthorityFromAccount(account);
+            if (!StringUtil.isEmpty(mAuthorityString)
```

Comment:
This looks to be copy/pasted out of `Authority` -- why can't we modify the behavior of that method to return `null` if the `tid` is blank? Line `797` would then set this value correctly

Reply:
Addressed.